### PR TITLE
Lesson 1 Update

### PR DESCRIPTION
**Pull request by dancortes94**
This PR updates the lesson 1 slides and solutions.
## Important
This PR contains work done in #31, take this into consideration.

### CODE REVIEW

File: lesson01/data.tf [hunk 1]
```diff
@@ -1,36 +1,38 @@
...
 data "aws_caller_identity" "current" {}
 
-data "aws_route53_zone" "current" {
-  name = "${var.domain}"
-}
+# Uncomment if you have a hosted zone on your AWS account
+# Gets hosted zone ID
+# data "aws_route53_zone" "current" {
+#   name = "${var.domain}"
+# }
```
Comment (estebancrw) [inline lesson01/data.tf:38]: styling: missing blank line at the end of the file

File: lesson01/main.tf [hunk 2]
```diff
@@ -1,11 +1,14 @@
...
-}
+# If you prefer, you can store the Terraform state in S3
+# https://www.terraform.io/docs/backends/types/s3.html
+# terraform {
+#   backend "s3" {
+#     region = "us-east-2"
+#   }
+# }
+
+# Add below your code
```
Comment (estebancrw) [inline lesson01/main.tf:14]: styling: missing blank line at the end of the file

File: lesson01/output.tf [hunk 3]
```diff
@@ -1,8 +1,14 @@
...
+
 # output "web_dns" {
-#   value = "${aws_route53_record.dns_web.name}"
+#   value = aws_instance.web.public_dns
 # }
 
+# Only uncomment if you have a hosted zone in Route53
+# output "web_dns" {
+#   value = aws_route53_record.dns_web.name
+# }
```
Comment (estebancrw) [inline lesson01/output.tf:14]: styling: missing blank line at the end of the file

File: lesson01/variables.tf [hunk 4]
```diff
@@ -4,27 +4,34 @@ variable "tags" {
...
+  description = "AWS key created to access the instance via ssh"
   type        = "string"
 }
+
+# Only uncomment if you have a hosted zone in Route53
+# variable "domain" {
+#   default     = "academy.wizeline.dev"
+#   description = "The domain name to use"
+#   type        = "string"
+# }
```
Comment (estebancrw) [inline lesson01/variables.tf:37]: styling: missing blank line at the end of the file

File: lesson01/versions.tf [hunk 5]
```diff
@@ -0,0 +1,3 @@
+terraform {
+  required_version = ">= 0.12"
+}
```
Comment (estebancrw) [inline lesson01/versions.tf:3]: styling: missing blank line at the end of the file

File: slides/lesson1/solution/main.tf [hunk 6]
```diff
@@ -40,29 +33,35 @@ resource "aws_security_group" "web" {
...
-  records = ["${aws_instance.web.public_ip}"]
-}
+# Uncomment this if you have a hosted zone in your AWS account
+# resource "aws_route53_record" "dns_web" {
+#   zone_id = "${data.aws_route53_zone.current.zone_id}"
+#   name    = "${data.aws_caller_identity.current.user_id}.${var.domain}"
+#   type    = "A"
+#   ttl     = 300
+#   records = ["${aws_instance.web.public_ip}"]
+# }
```
Comment (estebancrw) [inline slides/lesson1/solution/main.tf:67]: styling: missing blank line at the end of the file

File: slides/lesson1/solution/versions.tf [hunk 7]
```diff
@@ -0,0 +1,4 @@
+
+terraform {
+  required_version = ">= 0.12"
+}
```
Comment (estebancrw) [inline slides/lesson1/solution/versions.tf:4]: styling: missing blank line at the end of the file

File: slides/lesson1/solution/versions.tf [hunk 8]
```diff
@@ -0,0 +1,4 @@
+
```
Comment (estebancrw) [inline slides/lesson1/solution/versions.tf:1]: styling: let's remove the blank line

**Review by estebancrw — CHANGES_REQUESTED** [review]
Looks great! Just one blocking and one non-blocking issue.
blocking issue: since this PR is meant for lesson01 only, let's not modify lesson02 nor lesson03
non-blocking issue: a few styling changes - removing blank lines + adding blank lines at the end of the file

**Comment by dancortes94** [comment]
Closing in favor of #35

**Pull request closed without merging**